src/osd/PG.cc

```diff
@@ -3808,14 +3808,8 @@ void PG::replica_scrub(
     return;
   }
 
-  // compensate for hobject_t's with wrong pool from sloppy hammer OSDs
-  hobject_t start = msg->start;
-  hobject_t end = msg->end;
-  start.pool = info.pgid.pool();
-  end.pool = info.pgid.pool();
-
   build_scrub_map_chunk(
-    map, start, end, msg->deep, msg->seed,
+    map, msg->start, msg->end, msg->deep, msg->seed,
```

Comment:
I wasn't thinking that this code runs on the replicas, so now I see what you mean about interacting with older releases.  I'll put this back.